### PR TITLE
Cancel file Store during Abort

### LINKED ISSUE #1320: Terminate context when writer is aborted
When sstable writer is being aborted, it closes the local file but does nothing with the ongoing upload to s3.
This keeps the failing flow hanging until the request completes, but the result is meaningless.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1320

### CODE REVIEW

File: pyramid/file_test.go [hunk 1]
```diff
@@ -51,11 +51,13 @@ func TestPyramidWriteFile(t *testing.T) {
 
 	require.NoError(t, sut.Close())
 	require.NoError(t, sut.Store(ctx, filename))
+	require.NotNil(t, sut.cancelStore)
```
Comment (arielshaqed) [inline pyramid/file_test.go:54]: I don't understand what this tests.  AFAICS it only tests _that_ a cancel function was returned, not that it has anything to do with actually cancelling the context (e.g. correctly wired up).

File: pyramid/wr_file.go [hunk 2]
```diff
@@ -35,12 +36,21 @@ func (f *WRFile) Store(ctx context.Context, filename string) error {
 		return err
 	}
 
+	// keep the cancel function for the Store's context,
+	// so that the long operation is cancellable during a call to Abort.
+	ctx, f.cancelStore = context.WithCancel(ctx)
 	return f.store(ctx, filename)
 }
 
 // Abort delete the file and cleans all traces of it.
```
Comment (arielshaqed) [inline pyramid/wr_file.go:45]:
```suggestion
// Abort deletes the file and cleans all traces of it.
```

**Review by arielshaqed — APPROVED** [review]
Cool!
How can we test this (i.e., that cancellation is actually wired through & "works")?  It's OK to do it manually, even once, or in a unit test with a fake slow store, or even to open an issue.  Just let's make sure...

File: pyramid/file_test.go (hunk 1 above)
Comment (itaiad200) [reply pyramid/file_test.go:56]: Right, fixed.

**Pull request merged**